flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/SourceStreamTaskTest.java

```diff
@@ -275,13 +280,16 @@ public void finishingIgnoresExceptions() throws Exception {
 				SourceStreamTask::new,
 				BasicTypeInfo.STRING_TYPE_INFO);
 
+		final CompletableFuture<Void> waitingLatch = new CompletableFuture<>();
```

Comment:
`operatorRunningWaitingFuture`; wondering whether we should use a OneShotLatch instead; or whether CompFuture could completely subsume this.

Reply:
In this specific case it can. Normally the good thing about the latch is that you can reset it, which here we do not need.